cmd/cmd_utils.go

```diff
@@ -0,0 +1,200 @@
+					for i, step := range customCommandSteps {
+						// Prepare template data for arguments
+						argumentsData := map[string]string{}
+						for ix, arg := range customCommandArguments {
+							argumentsData[arg.Name] = args[ix]
+						}
+
+						// Prepare template data for flags
+						flags := cmd.Flags()
+						flagsData := map[string]string{}
+						for _, fl := range customCommandFlags {
+							if fl.Type == "" || fl.Type == "string" {
+								providedFlag, err := flags.GetString(fl.Name)
+								if err != nil {
+									u.PrintErrorToStdErrorAndExit(err)
+								}
+								flagsData[fl.Name] = providedFlag
+							}
+						}
+
+						// Prepare full template data
+						var data = map[string]map[string]string{
+							"Arguments": argumentsData,
+							"Flags":     flagsData,
+						}
+
+						// Parse and execute Go templates in the command's steps
+						t, err = template.New(fmt.Sprintf("step-%d", i)).Parse(step)
+						if err != nil {
+							u.PrintErrorToStdErrorAndExit(err)
+						}
+						var tpl bytes.Buffer
+						err = t.Execute(&tpl, data)
+						if err != nil {
+							u.PrintErrorToStdErrorAndExit(err)
+						}
+
+						// Prepare ENV vars
+						var envVarsList []string
+						for _, v := range customEnvVars {
+							key := v.Key
+							val := v.Value
+							valCommand := v.ValueCommand
+
+							if val != "" && valCommand != "" {
+								err = fmt.Errorf("either 'value' or 'valueCommand' can be specified for the ENV var, but not both.\n"+
+									"Custom command '%s %s' defines 'value=%s' and 'valueCommand=%s' for the ENV var '%s'",
+									parentCommand.Name(), commandConfig.Name, val, valCommand, key)
+								u.PrintErrorToStdErrorAndExit(err)
+							}
+
+							// If the command to get the value for the ENV var is provided, execute it
+							if valCommand != "" {
+								valCommandArgs := strings.Fields(valCommand)
+								res, err := e.ExecuteShellCommandAndReturnOutput(valCommandArgs[0], valCommandArgs[1:], ".", nil, false)
+								if err != nil {
+									u.PrintErrorToStdErrorAndExit(err)
+								}
+								val = res
+							} else {
+								// Parse and execute Go templates in the values of the command's ENV vars
+								t, err = template.New(fmt.Sprintf("env-var-%d", i)).Parse(val)
+								if err != nil {
+									u.PrintErrorToStdErrorAndExit(err)
+								}
+								var tplEnvVarValue bytes.Buffer
+								err = t.Execute(&tplEnvVarValue, data)
+								if err != nil {
+									u.PrintErrorToStdErrorAndExit(err)
+								}
+								val = tplEnvVarValue.String()
+							}
+
+							envVarsList = append(envVarsList, fmt.Sprintf("%s=%s", key, val))
+							err = os.Setenv(key, val)
+							if err != nil {
+								u.PrintErrorToStdErrorAndExit(err)
+							}
+						}
+
+						if c.Config.Logs.Verbose {
+							if len(envVarsList) > 0 {
+								u.PrintInfo("\nUsing ENV vars:")
+								for _, v := range envVarsList {
+									fmt.Println(v)
```

Comment:
Can we indent these so it's more readable ?

Reply:
I think it prints the ENV vars correctly and readable enough:

```
Using ENV vars:
ENV_VAR_1=ENV_VAR_1_value
ENV_VAR_2=ENV_VAR_2_value
```

Reply:
how would you like to improve it